### PR TITLE
add feature to replace jar after execution

### PR DESCRIPTION
**Pull request by ocadaruma**
Add `staging-jar` setting to deploy new jar while app is running.
This feature works as follows:

- If  both`app.jar` and `app.staging_jar` are set, replace jar file at `app.jar` with `app.staging_jar` after execution.

Motivation:

- Imagine that you run java-batch application via javactl.
- If you deploy new jar to `app.jar`, JVM class loader fails to load classes still unloaded.

### CODE REVIEW

File: src/javactl/executor/executor.py [hunk 1]
```diff
@@ -161,3 +162,13 @@ def _execute_commands(self, commands, now):
                         self.setting.app_setting.name, cmd, ret))
                     failed = True
         return self.copy(failed=failed)
+
+    def _replace_jar(self):
```
Comment (mogproject) [inline src/javactl/executor/executor.py:166]: I think this is necessary. (Do nothing when failed)
```
 if self.failed:
    return self
```

File: src/javactl/executor/executor.py [hunk 2]
```diff
@@ -161,3 +162,13 @@ def _execute_commands(self, commands, now):
...
+        jar = self.setting.app_setting.jar
+        staging_jar = self.setting.app_setting.staging_jar
+
+        if jar is not None and staging_jar is not None:
```
Comment (mogproject) [inline src/javactl/executor/executor.py:170]: You should implement "dry-run" mode.

**Review by mogproject — CHANGES_REQUESTED** [review]

**Comment by mogproject** [comment]
This task should be done by an external shell script and "post commands" option. Is there any reason why you do not want to use "post commands"?

Also, `staging-jar` option is not intuitive for me, and this feature is highly connected to a specific work flow. Therefore, I do not think I should add this feature to `javactl`.

**Comment by ocadaruma** [comment]
@mogproject Got it.

**Pull request closed without merging**